Dash-auth with dds 2.6

T4rk1n opened the pull request:
- Disable index overwrite and route wrapping if `DASH_LOGOUT_URL` environment variable is set.
- `PlotlyAuth.create_logout_button` returns a `dcc.LogoutButton` with the `DASH_LOGOUT_URL` if set.

Tests are still failing with timeouts.

- [x] test with 2.6

scjody commented:
@T4rk1n Thanks for the testing. I took a look at https://github.com/plotly/dash-deployment-server/pull/190 and it doesn't look like the redirect was added.

@tarzzz Can you please look into this? (Based on the design in https://github.com/plotly/streambed/issues/11817#issuecomment-439107597, a redirect is required so that `dash-auth` and `dcc` can send the user to the logout endpoint via a standard form `POST`.)

T4rk1n commented:
@tarzzz  If the redirect could take a url supplied in the form values it would be great. I could take it as prop and set hidden value input.

tarzzz commented:
@T4rk1n Would prefer to keep it simple at this point, to redirect to the DDS main page. It also makes sense because logging out from an app logs a user out from the DDS as well as all the other apps. 

We may add the redirect if it feels necessary.

T4rk1n commented:
Getting 405 on the revoke_token route for the old logout:

Invalidation failure HTTPError(u'405 Client Error: Method Not Allowed for url: https://plot.ly/Auth/o/revoke_token/',)

T4rk1n commented:
@scjody Please review.

We decided at the last dash meeting to deprecate `PlotlyAuth` and split `dash-auth` into 3 repos.

- `dash-enterprise-auth` for dds auth integrations, https://github.com/plotly/dash-enterprise-auth/pull/1, basically the logout button, get_username and the kerberos ticket method for now.
- `basic_auth.py` -> `dash-basic-auth`
- `oauth.py` -> `dash-oauth`

I added a pending deprecation warning to `PlotlyAuth.__init__`. I think dopsa should still use dash-auth with this PR patch to disable it if the `logout_url` is set to ease the transition.

The tests on circleci all get stuck with no output, they still pass locally.

scjody commented on dash_auth/plotly_auth.py:
```diff
@@ -21,6 +22,19 @@
 
+deprecation_notice = '''
+PlotlyAuth is being deprecated.
+If your app is still using dash-deployment-server < 2.6,
```
minor, but the product is called Dash Deployment Server (it's not a Python package).

scjody commented on dev-requirements.txt:
```diff
@@ -1,4 +1,5 @@
-dash_core_components
+# dash_core_components
+-e git+https://github.com/plotly/dash-core-components.git@logout-btn#egg=dash_core_components
```
Reminder to change this before merging if possible

T4rk1n commented:
Tested on 2.5 and 2.6, both works.
